api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineConfigurationManagerTest.java

```diff
@@ -16,182 +16,193 @@
-        dataStorageDao.createDataStorage(dataStorage);
         dataStorages.add(dataStorage);
 
-        S3bucketDataStorage bucketDataStorage = new S3bucketDataStorage(dataStorageDao.createDataStorageId(),
-                                                                        "testBucket", "test/path2");
+        S3bucketDataStorage bucketDataStorage = new S3bucketDataStorage(TEST_DS_S3_ID_1, "testBucket", "test/path2");
         bucketDataStorage.setOwner(TEST_OWNER1);
-        dataStorageDao.createDataStorage(bucketDataStorage);
         dataStorages.add(bucketDataStorage);
 
         // Data storages of user 2
-        dataStorage = new NFSDataStorage(dataStorageDao.createDataStorageId(), "testNFS2", "test/path3");
+        dataStorage = new NFSDataStorage(TEST_DS_NFS_ID_2, "testNFS2", "test/path3");
         dataStorage.setMountOptions("testMountOptions2");
         dataStorage.setOwner(TEST_OWNER2);
-        dataStorageDao.createDataStorage(dataStorage);
         dataStorages.add(dataStorage);
 
-        bucketDataStorage = new S3bucketDataStorage(dataStorageDao.createDataStorageId(),
-                                                                        "testBucket2", "test/path4");
+        bucketDataStorage = new S3bucketDataStorage(TEST_DS_S3_ID_2, "testBucket2", "test/path4");
         bucketDataStorage.setOwner(TEST_OWNER2);
-        dataStorageDao.createDataStorage(bucketDataStorage);
         dataStorages.add(bucketDataStorage);
 
-        dataStorages.forEach(ds -> aclTestDao.createAclForObject(ds));
-        aclTestDao.grantPermissions(dataStorage, TEST_OWNER1,
-                                    Collections.singletonList((AclPermission) AclPermission.READ));
+        //let's suppose, that we granted READ permission to both owners: TEST_OWNER1 and TEST_OWNER2
     }
 
     @Test
-    @Transactional(propagation = Propagation.REQUIRES_NEW)
-    @WithMockUser(username = TEST_OWNER1)
     public void testGetUnregisteredPipelineConfiguration() {
+        doReturn(TEST_DOCKER_IMAGE).when(pipelineVersionManager).getValidDockerImage(eq(TEST_IMAGE));
+        doReturn(dataStorages).when(dataStorageApiService).getWritableStorages();
+
         PipelineStart vo = getPipelineStartVO();
         PipelineConfiguration config = pipelineConfigurationManager.getPipelineConfiguration(vo);
-        Assert.assertFalse(config.getBuckets().isEmpty());
-        Assert.assertFalse(config.getNfsMountOptions().isEmpty());
+        assertFalse(config.getBuckets().isEmpty());
+        assertFalse(config.getNfsMountOptions().isEmpty());
 
         String[] buckets = config.getBuckets().split(";");
-
-        Assert.assertEquals(2, buckets.length);
+        assertEquals(4, buckets.length);
         for (String bucket : buckets) {
-            Assert.assertTrue(dataStorages.stream().anyMatch(ds -> bucket.equals(ds.getPathMask())));
+            assertTrue(dataStorages.stream().anyMatch(ds -> bucket.equals(ds.getPathMask())));
```

Comment:
I would prefer to have mode specific assertions (`anyMatch` is not accurate).

Reply:
Fixed